### ISSUE TITLE
Timeliner is winding down in favor of Timelinize

### ISSUE THREAD

**Issue by mholt** [mholt/timeliner#84]
As stated on this repository's README, this repo is no longer being actively developed, because its successor, [Timelinize](https://timelinize.com), is the future.

I hope this isn't disappointing to anyone. In fact, I want you to know how _awesome_ I think Timelinize is, so I want you to see it and try it for yourself.

A screenshot from a much earlier alpha version:

![image](https://github.com/mholt/timeliner/assets/1128849/95c4e448-d3d2-4822-94f3-dadbce9b9f80)

(That's right, there's finally a UI!)

Sometimes I tweet (xeet?) about neat aspects of the project, like how [we clean up noisy data from Google Location History, in some cases better than Google does](https://twitter.com/timelinize/status/1615508460954865664). 

If you're a developer, you can email me to request alpha access. **I only ask that as an early tester, you provide feedback and be engaged with our community during development.** (My email is on the linked website.) That's how we'll make the best possible software. :slightly_smiling_face:

Thank you! Let me know if you have any questions.

**Comment by diraneyya** [comment]
Is Timelinize open-source software?

**Comment by mholt** [comment]
Not currently. Maybe in the future. I haven't decided yet. Why?

**Comment by diraneyya** [comment]
All good. Timeliner was open-source with a very strict AGPL license, you also seem to be a very effective member of the open-source community, so one almost wonders if he/she is missing something when there is no link or mention of a GitHub repository in the successor to what started as an open-source project.

Yet, from what I understand, it is going to be a binary that the user would download, which then parses and displays the user's data, which is stored locally, so all in all it seems like a reasonable arrangment.

Best of luck and thank you for your great contributions. Caddy seems amazing I will surely test it in the next days.

**Comment by mholt** [comment]
Yeah, I strictly want to prevent Timeliner/Timelinize from being used for commercial gain, especially given how sensitive the data is that it processes. That's why it's strictly local-only.

Timelinize will not be a commercial venture even if it is closed source; the main thing I'm worried about is making it sustainable. It's important that a project like this lives on. I got lucky with Caddy as it has direct business use cases, but Timelinize does not; and I worry that after 7+ years of development, releasing it for free and open collaboration will only compound the burdens on my time.

That said, even if it is closed source, the data and its interoperability will always be very open. That is the nature of its design: local-only, SQLite index, plain files on disk. No encryption, no DRM, no uploading to a third party service, etc. Anyone will be able to query and manipulate their timeline with their own code. The schema is out in plain view. How the files are synced with the schema will be documented. Power users can go to town on it.

Open source has a lot of appealing aspects too, which I'm obviously aware of, and I'm heavily considering. And honestly, I would totally make it open source without question if I could just figure out how to compensate me for my time, given the consumer-oriented nature of this (as opposed to business focus). It really is a tool for me and my family, but is also generally useful and I think the world really needs this and is ready for it. So I want to make sure it gets out there one way or another.

I just want to avoid a situation where I've put it out there, then become so busy trying to make ends meet with other things (like Caddy; or if that loses its sponsorships, I'd have to get a separate full-time job, starving both projects), that it becomes stale.

It will likely be free while in public beta; then eventually I'd like to accept compensation for it. If donations can prove to be sustainable, great -- if not, I can require payment, and I'm not entirely sure how that'd look yet. Open to ideas/suggestions. (I have already given it a lot of thought, though.)

**Comment by diraneyya** [comment]
I think this is fair and I agree with you in that this is something that the world needs and is ready for.

I'm quite interested in the funding of open source projects and I think that I have found a good system for it. However, the system only works for open-source software with commercial uses.

A piece of software intended for offline personal use, on the other hand, seems to be reasonably best supported by charging a fee for use, as you suggested.

Besides, being a tool for viewing information locally I understand why making the source itself (rather than the data format or the schema) available/open is not necessarily a morally superior position to keeping it closed, under the assumption that there's no foul play or malware involved.

Hence thank you for the explanations and I wish you the best of luck with Caddy and this one.

**Comment by mholt** [comment]
Thank you! And thanks for understanding. :)

**Comment by mholt** [comment]
Actually, the last few days, I've been thinking seriously about this as it's been on my mind quite a bit.

Given my investment into this project over the years so far, and my desire for it to be sustainable, and being very particular for how it is designed/how it functions., I've had the mindset that keeping it closed source and charging for it when it is released and out of beta (see discussion above). But after thinking it over more, I am less convinced that is the path to the project's success.

If you think about "What is the point of keeping software closed source and charging for it?" it's to make money -- to pay bills, etc. To do business. I have a day job with Caddy through open source sponsorships because Caddy is used by businesses that rely on it. If I lose that source of income then I need to get a job or make a living with something else... possibly Timelinize? Timelinize has no such business use case, so I don't think it can appeal to companies and thus I don't think the sponsorship model will work. It's strictly consumer software, and it's not common for consumers to sponsor at significant amounts that can sustain a business or a family -- the software would have to be proprietary/paid. And then it would either have to be very expensive or reach tens of thousands of paying users, which is also not sustainable from a support perspective.

The point is, I think looking to make Timelinize a source of revenue, or to apply a business model that is expected to grow, is folly.

I also realize there are many in the community who have talents that can make the project better than I can, especially with data viz and UI.

Additionally, since this is client software, there are many more variables to consider as this will run in many more environments than I can test. Unlike Caddy, which runs on servers and has much simpler requirements, I've found that even though Timelinize works for me, getting it working on everyone's computers is much more difficult. Open source could be beneficial here.

What I might do instead is lean heavily on donations, since managing the project will still take considerable effort and I'll need the motivation as well as a gauge of interest to guide the project.

I would probably also carry over the AGPL license to try to mitigate commercial use of its code.

I like how other software seems to have success with the "payment recommended" model like WinRAR, Sublime Text, MakeMKV, and others -- so I'm not sure how it'll look yet, but if I do go to the open source route, maybe it'll be something like that (except those examples are closed source).

**Comment by teleclimber** [comment]
I sympathize with your struggle to balance sustainability and openness.

My 0.02: consider that instead of making payment recommended, you could make it payment required to download the signed executable binary. With the code open source, those who refuse to pay can build it themselves. Some might build and freely distribute a binary themselves, but it wouldn't be signed by you and your reputation has weight. Who knows what spyware is added to that executable?

I don't know what the conversion rate is for "payment recommended" approaches, but I suspect it's not great. By charging at the door, you could make the price quite low and come out ahead. (Maybe?)

One more thing: I think you'd want to get the trademark for Timelinize if you haven't already.

**Comment by mholt** [comment]
> My 0.02: consider that instead of making payment recommended, you could make it payment required to download the signed executable binary. With the code open source, those who refuse to pay can build it themselves. Some might build and freely distribute a binary themselves, but it wouldn't be signed by you and your reputation has weight. Who knows what spyware is added to that executable?

I actually did this exact thing with Caddy years ago (before v2). A lot of people fumed and expressed confusion over this, with so much pushback and hate for it that I actually almost quit the project.

> One more thing: I think you'd want to get the trademark for Timelinize if you haven't already.

I don't really like the name and am open to ideas for an alternative...

**Comment by teleclimber** [comment]
> I actually did this exact thing with Caddy years ago (before v2). A lot of people fumed and expressed confusion over this, with so much pushback and hate for it that I actually almost quit the project.

Oh no, that's brutal. I wonder if some of that might be due to the fact the alternatives are freely downloadable (Apache, nginx, etc...). But in any case, I understand if you wouldn't want to repeat that experiment!
